plugin/db/mysql/mysql.go

```diff
@@ -913,21 +914,30 @@ func (driver *Driver) Dump(ctx context.Context, database string, out io.Writer,
 }
 
 // Restore restores a database.
-func (driver *Driver) Restore(ctx context.Context, sc *bufio.Scanner) (err error) {
+func (driver *Driver) Restore(ctx context.Context, sc *bufio.Scanner, config db.RestoreConfig) (err error) {
 	txn, err := driver.db.BeginTx(ctx, nil)
 	if err != nil {
 		return err
 	}
 	defer txn.Rollback()
 
-	f := func(stmt string) error {
+	fnExecuteStmt := func(stmt string) error {
```

Comment:
Would this be more readable to add it here?


if config.IsGhostTable {
  stmt = rewriteToGhostTableForRestore(stmt)
}

Reply:
I'm afraid not. The `stmt` is passed in to `fnExecuteStmt` in a callback style in `util.ApplyMultiStatements` after parsing the SQL text. We are replacing the callback function here if `IsGhostTable` is true.

Reply:
IIUC, L932 ~ 938 is to rewrite the query if it's ghost table, other than that the logic is the same as the non-ghost case. So I am thinking of removing that section and use a single callback

fnExecuteStmt := func(stmt string) error {
                if config.IsGhostTable {
                   stmt = rewriteToGhostTableForRestore(stmt)
                }
		if _, err := txn.Exec(stmt); err != nil {
			return err
		}
		return nil
}